testing/scala/APKBUILD

```diff
@@ -0,0 +1,111 @@
+# Contributor: norbjd
+# Maintainer:
+pkgname=scala
+pkgver=2.12.2
+pkgrel=0
+pkgdesc="Scala 2.12.2"
+url="http://scala-lang.org"
+arch="all"
+license="BSD3"
+
+# bash          : needed for scripts (e.g. scala)
+# openjdk8-jre  : needed for execution 
+depends="bash openjdk8-jre"
+
+# openjdk8      : needed for sbt + compilation
+# sbt           : needed for compilation
+# graphviz      : needed for diagrams generation in scaladoc
+makedepends="openjdk8 sbt graphviz"
+
+# documentation is separated
+subpackages="${pkgname}-doc"
+
+source="${pkgname}-${pkgver}.tar.gz::https://github.com/scala/scala/archive/v2.12.2.tar.gz"
+builddir="${srcdir}/${pkgname}-${pkgver}"
+
+# important directories
+_scala_home="/usr/share/${pkgname}-${pkgver}"
+_scala_docs_home="/usr/share/doc/${pkgname}-${pkgver}"
+_scala_man_home="/usr/share/man/man1"
+_bin_home="/usr/bin"
+
+# sbt command with JVM options (memory)
+_sbt() {
+        sbt -Xms2G -Xmx4G -XX:MaxMetaspaceSize=1G "$@"
+}
+
+build() {
+        cd "$builddir"
+
+        # use sbt to compile, package, and create doc
+	_sbt ';dist/mkPack;doc;compile:managedResources;packageDoc' || return 1
+}
+
+check() {
+        cd "$builddir"
+
+        # check if all scripts have been created and work
+	./build/pack/bin/scala -version
+	./build/pack/bin/scalac -version
+	./build/pack/bin/scaladoc -version
+	./build/pack/bin/scalap -version
+	./build/pack/bin/fsc -version
+}
+
+package() {
+        cd "$builddir"
+
+        # copy scripts
+	mkdir -p "$pkgdir"/"$_scala_home" || return 1
+	cp -R build/pack/* "$pkgdir"/"$_scala_home"/ || return 1
+
+        # remove useless specific Windows scripts
+	rm "$pkgdir"/"$_scala_home"/bin/*.bat || return 1
+
+        # create symbolic links
+	mkdir -p "$pkgdir"/"$_bin_home" || return 1
+	for _bin in `ls "$pkgdir"/"$_scala_home"/bin/`
```

Comment:
I'd use local variable here: `local bin; for bin in ...`. Also, you can use `for bin in "$pkgdir"/"$scala_home"/bin/*` instead of a subprocess `ls` invocation.